src/database/database-files.cpp

```diff
@@ -177,3 +178,106 @@ void PlayerDatabaseFiles::listPlayers(std::vector<std::string> &res)
 		res.emplace_back(player.getName());
 	}
 }
+
+AuthDatabaseFiles::AuthDatabaseFiles(const std::string &savedir) : m_savedir(savedir)
+{
+	readAuthFile();
```

Comment:
You want the whole thing to be from a constructor that gets called multiple times when an object is made?

Try something sane and have `AuthDatabaseFiles` as a global object and call `readAuthFile()` directly. For example:

( Just an example. don't use this code, it's broken )
```
extern std::unique_ptr<AuthDatabaseFiles> sqlauth; // Declare the global object. Should be in a header that does global stuff. `std::unique_ptr` to prevent memory leaks.

std::unique_ptr<AuthDatabaseFiles> sqlauth = nullptr; // In a .cpp file for the header mentioned above.

// In this file, you can just have something call `readAuthFile()` by doing...
sqlauth = std::unique_ptr<AuthDatabasefiles>( new AuthDatabaseFiles() );
sqlauth->readAuthFile();
```

Boom.

Reply:
The `AuthDatabase` is created once per world, when the server environment is created, so I don't think it's unreasonable to read the file in the constructor.

I do seem to be missing a delete, though, so I'll either add that or more likely use a smart pointer in the serverenvironment.

Reply:
I added a `delete`, on par with the player database a few lines above.